Source/JavaScriptCore/yarr/YarrInterpreter.cpp

```diff
@@ -1287,16 +1494,59 @@ class Interpreter {
         return true;
     }
 
-#define MATCH_NEXT() { ++context->term; goto matchAgain; }
-#define BACKTRACK() { --context->term; goto backtrack; }
+#define MATCH_NEXT() { if (verbose) { dataLog(" - Match\n"); if (btrack) { dataLog("  Matching\n"); btrack = false; } }  ++context->term; goto matchAgain; }
+#define BACKTRACK() { if (verbose) { dataLog(" - Backtrack\n"); if (!btrack) { dataLog("  Backtracking\n"); btrack = true; } } --context->term; goto backtrack; }
 #define currentTerm() (disjunction->terms[context->term])
+
+#define DUMP_TERM() \
+{ \
+    if (verbose) { \
+        termDumper.dumpTerm(context->term, currentTerm()); \
+        dataLog(" pos:", input.getPos()); \
+    } \
+}
+
+#define DUMP_EXTRA(...) \
+{ \
+    dataLogIf(verbose, " ", __VA_ARGS__); \
+}
+
+#define DUMP_EXTRA_IF(predicate, ...) \
+{ \
+    dataLogIf(verbose && predicate, " ", __VA_ARGS__); \
+}
+
+#define DUMP_CURR_CHAR() \
+{ \
+    if (verbose) { \
+        unsigned offset = currentTerm().inputPosition; \
+        if (currentTerm().matchDirection() == Backward && currentTerm().atom.quantityType == QuantifierType::FixedCount) \
+            offset -= currentTerm().atom.quantityMaxCount - 1; \
+        dataLog(" off:", offset, " got:'"); \
+        int ch = -1; \
+        if (input.isValidNegativeInputOffset(offset)) \
+            ch = input.readChecked(offset); \
+        if (ch == -1) \
+            dataLog("<illegal>"); \
+        else if (ch < 0x1000)\
+            dataLog(static_cast<wchar_t>(ch)); \
```

Comment:
Let's use `char16_t` instead.

Reply:
We don't have a printStream method for `char16_t`, only `wchar_t`.  I did add a `0` to the comparison value on the line above so that we handle all BMP characters.

Reply:
Changed the printInternal() method and this use to be char16_t.